src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/vmware/VMWareProvider.java

```diff
@@ -57,33 +61,99 @@
             String instanceImageId = instance.getImage().split("/")[1];
 
             Folder rootFolder = vmWareServiceInstanceCache.getServiceInstance(infrastructure).getRootFolder();
-
-            VirtualMachineCloneSpec vmcs = new VirtualMachineCloneSpec();
-
-            vmcs.setLocation(vmWareProviderVirualMachineUtil.getVirtualMachineRelocateSpec(rootFolder));
-            vmcs.setPowerOn(true);
-            vmcs.setTemplate(false);
-            vmcs.setConfig(getVirtualMachineConfigSpec(instance));
-
-            Folder vmFolder = vmWareProviderVirualMachineUtil.searchFolderByName(instanceFolder, rootFolder);
+            Folder vmFolder = vmWareProviderVirtualMachineUtil.searchFolderByName(instanceFolder, rootFolder);
+            VirtualMachine vmToClone = vmWareProviderVirtualMachineUtil
+                    .searchVirtualMachineByName(instanceImageId, rootFolder);
 
             return IntStream.rangeClosed(1, Integer.valueOf(instance.getNumber()))
-                    .mapToObj(i -> cloneVM(instanceImageId, instance, rootFolder, vmcs, vmFolder))
+                    .mapToObj(instanceIndexStartAt1 -> cloneVM(vmToClone, createUniqInstanceTag(instance.getTag(),
+                            instanceIndexStartAt1), instance, rootFolder,
+                            createVirtualMachineCloneSpec(instanceIndexStartAt1, vmToClone, instance),
+                            vmFolder))
                     .map(vm -> instance.withId(vm.getConfig().getUuid())).collect(Collectors.toSet());
 
         } catch (RemoteException e) {
-            logger.error("ERROR when creating VMWare istance with : " + instance, e);
-            throw new RuntimeException("ERROR when creating VMWare istance with : " + instance, e);
+            logger.error("ERROR when creating VMWare instance with : " + instance, e);
+            throw new RuntimeException("ERROR when creating VMWare instance with : " + instance, e);
+        }
+
+    }
+
+    /**
+     * Create a uniq tag for a VM based on the original tag provided and the instance index
+     *
+     * @param tagBase       the tag base
+     * @param instanceIndex the instance index
+     * @return a uniq VM tag
+     */
+    private String createUniqInstanceTag(String tagBase, int instanceIndex) {

```

Comment:
Since it does not access to any class attributes, why not considering it as a private static class?